src/components/join-flow/join-flow-steps.jsx

```diff
@@ -22,14 +93,81 @@ class UsernameStep extends React.Component {
                     password: '',
                     passwordConfirm: ''
                 }}
+                validate={this.validateForm}
                 validateOnBlur={false}
                 validateOnChange={false}
+                onSubmit={this.handleValidSubmit}
             >
-                <JoinFlowStep
-                    description={this.props.intl.formatMessage({id: 'registration.usernameStepDescription'})}
-                    title={this.props.intl.formatMessage({id: 'general.joinScratch'})}
-                    waiting={this.props.waiting}
-                />
+                {props => {
+                    const {
+                        errors,
+                        handleSubmit,
+                        isSubmitting,
+                        validateField,
+                        values
+                    } = props;
+                    return (
+                        <JoinFlowStep
+                            description={this.props.intl.formatMessage({id: 'registration.usernameStepDescription'})}
+                            title={this.props.intl.formatMessage({id: 'general.joinScratch'})}
+                            waiting={isSubmitting}
+                            onSubmit={handleSubmit}
+                        >
+                            <div>
+                                <div className="username-label">
+                                    <b>
+                                        {this.props.intl.formatMessage({id: 'registration.createUsername'})}
+                                    </b>
+                                </div>
+                                <FormikInput
+                                    className={errors.username ? 'fail' : ''}
+                                    error={errors.username}
+                                    id="username"
+                                    name="username"
+                                    validate={this.validateUsernameIfPresent}
+                                    onBlur={() => validateField('username')} // eslint-disable-line react/jsx-no-bind
+                                />
+                                <b>
+                                    {this.props.intl.formatMessage({id: 'general.password'})}
+                                </b>
+                                <div
+                                    onClick={this.handleChangeShowPassword}
+                                >
+                                    {this.state.showPassword ? 'Hide password' : (
```

Comment:
I'm I'm reading correctly "Hide Password" is just some text we want to show that we don't have a message for yet? Maybe add a TODO to turn it into a message?

Reply:
Good idea!